client/lib/plans/test/plan-lookups.js

```diff
@@ -0,0 +1,580 @@
+		expect( isBusinessPlan( PLAN_JETPACK_PERSONAL ) ).to.equal( false );
+		expect( isBusinessPlan( PLAN_JETPACK_PERSONAL_MONTHLY ) ).to.equal( false );
+		expect( isBusinessPlan( PLAN_PREMIUM ) ).to.equal( false );
+		expect( isBusinessPlan( PLAN_JETPACK_PREMIUM ) ).to.equal( false );
+	} );
+} );
+
+describe( 'getMonthlyPlanByYearly', () => {
+	test( 'should return a proper plan', () => {
+		expect( getMonthlyPlanByYearly( PLAN_JETPACK_PERSONAL ) ).to.equal(
+			PLAN_JETPACK_PERSONAL_MONTHLY
+		);
+		expect( getMonthlyPlanByYearly( PLAN_JETPACK_PREMIUM ) ).to.equal(
+			PLAN_JETPACK_PREMIUM_MONTHLY
+		);
+		expect( getMonthlyPlanByYearly( PLAN_JETPACK_BUSINESS ) ).to.equal(
+			PLAN_JETPACK_BUSINESS_MONTHLY
+		);
+		expect( getMonthlyPlanByYearly( PLAN_JETPACK_FREE ) ).to.equal( '' );
+		expect( getYearlyPlanByMonthly( 'unknown_plan' ) ).to.equal( '' );
+	} );
+} );
+
+describe( 'getYearlyPlanByMonthly', () => {
+	test( 'should return a proper plan', () => {
+		expect( getYearlyPlanByMonthly( PLAN_JETPACK_PERSONAL_MONTHLY ) ).to.equal(
+			PLAN_JETPACK_PERSONAL
+		);
+		expect( getYearlyPlanByMonthly( PLAN_JETPACK_PREMIUM_MONTHLY ) ).to.equal(
+			PLAN_JETPACK_PREMIUM
+		);
+		expect( getYearlyPlanByMonthly( PLAN_JETPACK_BUSINESS_MONTHLY ) ).to.equal(
+			PLAN_JETPACK_BUSINESS
+		);
+		expect( getYearlyPlanByMonthly( 'unknown_plan' ) ).to.equal( '' );
+	} );
+} );
+
+describe( 'getPlanClass', () => {
+	test( 'should return a proper class', () => {
+		expect( getPlanClass( PLAN_FREE ) ).to.equal( 'is-free-plan' );
+		expect( getPlanClass( PLAN_JETPACK_FREE ) ).to.equal( 'is-free-plan' );
+		expect( getPlanClass( PLAN_PERSONAL ) ).to.equal( 'is-personal-plan' );
+		expect( getPlanClass( PLAN_PERSONAL_2_YEARS ) ).to.equal( 'is-personal-plan' );
+		expect( getPlanClass( PLAN_JETPACK_PERSONAL ) ).to.equal( 'is-personal-plan' );
+		expect( getPlanClass( PLAN_JETPACK_PERSONAL_MONTHLY ) ).to.equal( 'is-personal-plan' );
+		expect( getPlanClass( PLAN_PREMIUM ) ).to.equal( 'is-premium-plan' );
+		expect( getPlanClass( PLAN_PREMIUM_2_YEARS ) ).to.equal( 'is-premium-plan' );
+		expect( getPlanClass( PLAN_JETPACK_PREMIUM ) ).to.equal( 'is-premium-plan' );
+		expect( getPlanClass( PLAN_JETPACK_PREMIUM_MONTHLY ) ).to.equal( 'is-premium-plan' );
+		expect( getPlanClass( PLAN_BUSINESS ) ).to.equal( 'is-business-plan' );
+		expect( getPlanClass( PLAN_BUSINESS_2_YEARS ) ).to.equal( 'is-business-plan' );
+		expect( getPlanClass( PLAN_JETPACK_BUSINESS ) ).to.equal( 'is-business-plan' );
+		expect( getPlanClass( PLAN_JETPACK_BUSINESS_MONTHLY ) ).to.equal( 'is-business-plan' );
+	} );
+} );
+
+describe( 'getPlan', () => {
+	test( 'should return a proper plan - by key', () => {
+		expect( getPlan( PLAN_PERSONAL ) ).to.equal( PLANS_LIST[ PLAN_PERSONAL ] );
+	} );
+
+	test( 'should return a proper plan - by value', () => {
+		expect( getPlan( PLANS_LIST[ PLAN_PERSONAL ] ) ).to.equal( PLANS_LIST[ PLAN_PERSONAL ] );
```

Comment:
I'm curious about this use case, why would the caller ask for a plan object when it already has one?

Reply:
AFAIR I was worried about places when argument was named `plan` and it wasn't clear if it's universally string or object - but I think we ended up clearing things up and this usage was never utilized.

Reply:
Got it, we can leave it for a future janitorial, probably worth doing a usage audit, and simplifying the function here.